### PR TITLE
Update to java 16, javafx 16 and spring boot 2.5.0 

### LINKED ISSUE #2298: Support Java 16+
**Is your feature request related to a problem? Please describe.**
Currently, when running the client on linux, it requires java version 15 *exactly*.

**Describe the solution you'd like**
Arch (my distro) recently upgraded to java 16. Knowing the commitment to backward compatibility of java, I have no reason to suspect java 16 cannot run the client.
So perhaps instead of checking for version 15, check for version 15 or higher?

Client version: 1.4.6, gotten from the published released zip file on github

### PR DESCRIPTION
**Pull request by Sheikah45**
Also remove min java version

Fixes #2298

### CODE REVIEW

**Comment by Sheikah45** [comment]
Hmm tests pass locally for me

**Comment by dev-nulled** [comment]
We just bumped up to Springboot 2.5.2 at our shop.  (not sure if we care about the minor version, but as long as it was going 2.5.x...)

**Comment by Sheikah45** [comment]
yeah I just wasn't aware 2.5.2 existed. Although the biggest thing is that this isnt exporting the modules properly in checks

File: build.gradle [hunk 1]
```diff
@@ -316,10 +317,10 @@ dependencies {
...
 
   annotationProcessor("org.springframework.boot:spring-boot-configuration-processor")
-  annotationProcessor("org.projectlombok:lombok")
-  testAnnotationProcessor("org.projectlombok:lombok")
-  compileOnly("org.projectlombok:lombok")
-  testCompileOnly("org.projectlombok:lombok")
+  annotationProcessor("org.projectlombok:lombok:${lombokVersion}")
+  testAnnotationProcessor("org.projectlombok:lombok:${lombokVersion}")
+  compileOnly("org.projectlombok:lombok:${lombokVersion}")
+  testCompileOnly("org.projectlombok:lombok:${lombokVersion}")
```
Comment (Brutus5000) [inline build.gradle:323]: Use the latest Spring Boot 2.5.2 and you have the latest lombok. We should try to use the lombok bundled with spring boot if possible.

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (Sheikah45) [reply build.gradle:323]: Yeah this was an attempt of me hoping something else was causing the not exported errors and trying different things to resolve it. I will change it back.

**Comment by Sheikah45** [comment]
Going to update things separately

**Pull request closed without merging**